fcl/gen/corsika/prodcorsika_proton_intime_icarus.fcl

```diff
@@ -0,0 +1,109 @@
+#include "corsika_icarus.fcl"
+#include "services_icarus.fcl"
+#include "services_icarus_simulation.fcl"
+#include "largeantmodules_icarus.fcl"
+#include "photpropservices_icarus.fcl"
+
+
+process_name: CosmicsCorsikaCMCGenAndG4InTime
+
+services:
+{
+  #FileCatalogMetadata:  @local::art_file_catalog_mc
+  @table::icarus_gen_services
+  @table::icarus_g4_services 
+}
+
+#Start each new event with an empty event.
+source:
+{
+  module_type:     EmptyEvent
+  timestampPlugin: { plugin_type: "GeneratedEventTimestamp" }
+  maxEvents:       10          # Number of events to create
+  firstRun:        1           # Run number to use for this file
+  firstEvent:      1           # number of first event in the file
+}
+
+# Define and configure some modules to do work on each event.
+# First modules are defined; they are scheduled later.
+# Modules are grouped by type.
+physics:
+{
+
+ producers:
+ {
+   cosmgen: @local::icarus_corsika_p
```

Comment:
Yes -- I noticed this when I tried to use MC with this file in other stages.

I chose the name "cosmgen" to keep the label of the Corsika producer consistent with Corsika generation in BNB+Cosmic MC. It was my preference to keep the name of each type of generator consistent rather than keeping a single consistent name for the "primary" generator in each type of MC.

However, if the preference is the other way around in ICARUS, I can change this.

Reply:
Given the content, it makes more sense in my mind to be consistent with the cosmic ray generation configuration (as opposed to cosmic rays + beam). ICARUS configuration is `fcl/gen/corsika/prodcorsika_standard_icarus.fcl`, which turns out to have a generator label `generator`.
I don't have strong opinion on that, though, and if you in the end prefer `cosmgen` I am fine with that too.

Reply:
Maybe more relevant is to specify in the FHiCL file name which beam the generation is in time with.
The custom in ICARUS is currently to append the beam name: `prodcorsika_proton_intime_icarus_bnb.fcl`.

Reply:
I'm happy to make those changes -- changed both filename and generator name in the last commit.